src/components/PopoverMenu.tsx

```diff
@@ -314,6 +314,13 @@ function PopoverMenu({
         setCurrentMenuItems(menuItems);
     }, [menuItems]);
 
+    useEffect(() => {
+        if (isVisible) {
+            return;
+        }
+        setFocusedIndex(currentMenuItemsFocusedIndex);
+    }, [isVisible, currentMenuItemsFocusedIndex, setFocusedIndex]);
```

Comment:
`currentMenuItemsFocusedIndex` is changed due to the change in `currentMenuItems` which is due to the call `setCurrentMenuItems(menuItems)` that is done in the above `useLayoutEffect`. This is a known change and not a side effect and it should be done as soon as possible, i.e. call `setFocusedIndex` right after `setCurrentMenuItems`.

Note: Keep the isVisible condition but add a comment that we don't want the focus index to change if we dynamically change the items e.g.

If we have:

- Item 1 (Selected)
- Item 2 (Focused)
- Item 3

then the selected item is changed (externally), the result should be as follow:

- Item 1
- Item 2 (Focused)
- Item 3 (Selected)

Reply:
Added comment with the help from ChatGPT.

Reply:
Can you move the setFocusedIndex call as well

Reply:
Sorry, move to where? And why? I don't quite understand your first comment actually 😅

Reply:
Move it to the above useLayoutEffect

Reply:
Oh, I get what you're trying to say now. But if we do that, then we shouldn't add isVisible to the useLayoutEffect deps right?

```
useLayoutEffect(() => {
    if (menuItems.length === 0) {
        return;
    }
    setEnteredSubMenuIndexes(CONST.EMPTY_ARRAY);
    setCurrentMenuItems(menuItems);
    if (isVisible) {
        return;
    }
    setFocusedIndex(currentMenuItemsFocusedIndex);
// eslint disable
}, [menuItems, currentMenuItemsFocusedIndex, setFocusedIndex]);
```

Also, I don't think this would work. Why? When we call `setCurrentMenuItems(menuItems);`, `currentMenuItemsFocusedIndex` is not updated yet, so calling `setFocusedIndex(currentMenuItemsFocusedIndex);` right after it won't work. It'll be updated correctly after the 2nd trigger of `useLayoutEffect`.

Reply:
Got it now, updated